### ISSUE TITLE
RabbitMQ listener retry limitless

### ISSUE THREAD

**Issue by zhanjiejun** [spring-projects/spring-boot#7631]
```
@Service
@RabbitListener(queues = { Constants.APPROVAL_CALLBACK_TOPIC })
public class ApprovalConsumer {

	@RabbitHandler
	public void process(@Payload Map<String, Object> map) {
		throw new RuntimeException();
	}

}
```

when the method of my consumer throw a exception, the message will be delivered to my consumer limitless

and the console print the exception log over and over again

```
org.springframework.amqp.rabbit.listener.exception.ListenerExecutionFailedException: Listener method 'public void com.hikvision.oap.consumer.ApprovalConsumer.process(java.util.Map<java.lang.String, java.lang.Object>)' threw exception
	at org.springframework.amqp.rabbit.listener.adapter.MessagingMessageListenerAdapter.invokeHandler(MessagingMessageListenerAdapter.java:138)
	at org.springframework.amqp.rabbit.listener.adapter.MessagingMessageListenerAdapter.onMessage(MessagingMessageListenerAdapter.java:105)
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:777)
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:700)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.access$001(SimpleMessageListenerContainer.java:95)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$1.invokeListener(SimpleMessageListenerContainer.java:187)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.invokeListener(SimpleMessageListenerContainer.java:1187)
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.executeListener(AbstractMessageListenerContainer.java:681)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.doReceiveAndExecute(SimpleMessageListenerContainer.java:1165)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.receiveAndExecute(SimpleMessageListenerContainer.java:1149)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.access$1100(SimpleMessageListenerContainer.java:95)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.run(SimpleMessageListenerContainer.java:1312)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: null

```

the default value of property "retry enable" is false, but it's useless
and if i enable retry of listener, it will retry 3 times as the property "max-attempts" is setted

**Comment by wilkinsona** [comment]
> the message will be delivered to my consumer limitless

As I understand it, this is standard Spring AMQP and Rabbit MQ behaviour. What did you expect to happen?

> the default value of property "retry enable" is false, but it's useless

Why do you consider the property to be useless? It controls whether or not the listener is configured with a retry interceptor. When set to `false` (the default) no retry interceptor is configured. When set to `true` a retry interceptor is configured allowing you to configure retry behaviour.

/cc @garyrussell

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by spring-issuemaster** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.

**Comment by mshetty794** [comment]
I am facing the same issue where i have configured all the properties, but i see retry goes limitless
below is the my configuration in yaml file 

listener:
        type: simple
        simple: 
          retry:
            enabled: true
            max-attempts: 3
            max-interval: 60000
            multiplier: 1
          acknowledge-mode: manual
          prefetch: 1

**Comment by zhanjiejun** [comment]
Your issue is different from mine.
The issue I described is that:
**A)** I set retry enabled 'false', the message will be delivered to my consumer limitless. But I think the message should be delivered for just once when consumer throw a exception, and no retry (Maybe it's not retry by spring-boot, just rabbitmq doesn't get a reply, so the message is delivered limitless).
**B)** I set retry enabled ‘true‘ and max-attempts ‘3‘, and the consumer receive message for 3 times.

At that time, my spring-boot version is 1.5.x, and the config is correct when retry enabled set true.
Maybe in new version 2.1.x, more config is needed.
